src/utils/articles_api.js

```diff
@@ -21,3 +21,47 @@ export const getArticleComments = (article_id) => {
     return response.data.comments;
   });
 };
+
+export const likeArticle = (article_id, setArticle) => {
+  setArticle((currentArticle) => {
+    currentArticle += 1;
+    return currentArticle;
+  });
+  return articlesAPI
+    .patch(`/${article_id}`, { inc_votes: 1 })
+    .then((response) => {
+      console.log(response.data.article.votes);
+      return response.data.article.votes;
+    })
+    .catch((err) => {
+      console.log(err);
+      let error = document.getElementById(`article-vote-error-${article_id}`);
+      error.innerText = " - Serverside error, please try again later";
+      setArticle((currentArticle) => {
+        currentArticle.votes -= 1;
+        return currentArticle;
+      });
+    });
+};
+
+export const dislikeArticle = (article_id, setArticle) => {
+  setArticle((currentArticle) => {
```

Comment:
You will need to update the votes to make sure they are rendered optimistically. Also it should be done in the component.